apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/FissClaimTransformerV2.java

```diff
@@ -112,132 +196,279 @@ private static CodeableConcept getPriority() {
-                            new Coding(
-                                "http://hl7.org/fhir/sid/icd-10",
-                                claimGroup.getAdmitDiagCode(),
-                                null))))
-            .setType(
-                Collections.singletonList(
-                    new CodeableConcept()
-                        .setCoding(
-                            Collections.singletonList(
-                                new Coding(
-                                    ExDiagnosistype.ADMITTING.getSystem(),
-                                    ExDiagnosistype.ADMITTING.toCode(),
-                                    ExDiagnosistype.ADMITTING.getDisplay()))))));
-  }
-
-  private static List<Claim.ProcedureComponent> getProcedure(PreAdjFissClaim claimGroup) {
-    List<Claim.ProcedureComponent> procedure = new ArrayList<>();
-
-    // Sort proc codes by priority prior to building resource
-    List<PreAdjFissProcCode> procCodes = new ArrayList<>(claimGroup.getProcCodes());
-    procCodes.sort(Comparator.comparingInt(PreAdjFissProcCode::getPriority));
-
-    for (int i = 0; i < procCodes.size(); ++i) {
-      PreAdjFissProcCode procCode = procCodes.get(i);
-      Claim.ProcedureComponent component =
-          new Claim.ProcedureComponent()
-              .setSequence((i + 1))
-              .setDate(
-                  procCode.getProcDate() == null
-                      ? null
-                      : Date.from(
-                          procCode.getProcDate().atStartOfDay(ZoneId.systemDefault()).toInstant()))
-              .setProcedure(
-                  new CodeableConcept()
-                      .setCoding(
-                          Collections.singletonList(
-                              new Coding(
-                                  "http://hl7.org/fhir/sid/icd-10",
-                                  procCode.getProcCode(),
-                                  null))));
-
-      procedure.add(component);
+                    new CodeableConcept(
+                        new Coding(
+                            C4BBIdentifierType.NPI.getSystem(),
+                            C4BBIdentifierType.NPI.toCode(),
+                            C4BBIdentifierType.NPI.getDisplay())))
+                .setSystem(TransformerConstants.CODING_NPI_US)
+                .setValue(claimGroup.getNpiNumber()));
+      }
+
+      if (claimGroup.getLobCd() != null) {
+        reference.setExtension(
+            List.of(
+                new Extension(BBCodingSystems.CLM_FACILITY_TYPE_CODE)
+                    .setValue(
+                        new Coding(
+                            BBCodingSystems.CLM_FACILITY_TYPE_CODE, claimGroup.getLobCd(), null))));
+      }
+
+      return reference;
     }
 
-    return procedure;
+    return null;
+  }
+
+  private static List<Claim.DiagnosisComponent> getDiagnosis(
+      PreAdjFissClaim claimGroup, boolean isIcd9) {
+    final String systemSuffix = isIcd9 ? "9-cm" : "10";
```

Comment:
These magic values could use a descriptive constant so they can be better described.

Reply:
It's just which system to use, I could swap the ternary to do the entire coding system instead of the suffix

Reply:
If it let you use a more descriptive (and existing) ICD9/10 code constant, I'd recommend changing the ternary to use that just for readability and understanding at a glance.

Reply:
Updated the ternary to switch the entire system url using existing references